numpy/core/shape_base.py

```diff
@@ -348,3 +353,83 @@ def stack(arrays, axis=0):
     sl = (slice(None),) * axis + (_nx.newaxis,)
     expanded_arrays = [arr[sl] for arr in arrays]
     return _nx.concatenate(expanded_arrays, axis=axis)
+
+
+def block(tup_tup):
+    """
+    Create block arrays similar to Matlab's "square bracket stacking":
+
+            [A A; B B]
+
+    You can create a block array with the same notation you use for
+    `np.array`.
+
+    Parameters
+    ----------
+    tup_tup : sequence of sequence of ndarrays
```

Comment:
I didn't want to use `*args` because a) I don't like `*args`, b) looking at the other pull requests here `*args` seems to be discouraged, c) you can't add any other arguments (excecpt `**kwargs` which is also not great) and on the mailinglist there was some discussion on additotional parameters, and d) block behaves exactly like `array` and everybody is familiar with that. But I'm open for changes.

I agree, `tup_tup` can have a better name.